src/coreclr/jit/gentree.cpp

```diff
@@ -19468,6 +19468,19 @@ GenTree* Compiler::gtNewSimdBinOpNode(genTreeOps  op,
                     intrinsic = NI_AVX2_Add;
                 }
             }
+            else if (simdSize == 64)
+            {
+                if (varTypeIsByte(simdBaseType) || varTypeIsShort(simdBaseType))
```

Comment:
This is slightly more efficient:
```suggestion
                if (varTypeIsSmall(simdBaseType))
```

It technically also includes a check for `TYP_BOOL`, but higher up checks (like `varTypeIsArithmetic`) will have ensured that it can't be bool.